tt_metal/tt_metal.cpp

```diff
@@ -439,7 +439,8 @@ namespace detail {
             for (const auto &logical_core : logical_cores) {
                 launch_msg_t *msg = &program.kernels_on_core(logical_core)->launch_msg;
                 auto physical_core = device->physical_core_from_logical_core(logical_core, core_type);
-                not_done_cores.insert(physical_core);
+                std::pair< std::unordered_set<CoreCoord>::iterator ,bool> status =  not_done_cores.insert(physical_core);
```

Comment:
Can you show that this gives performance benefits with data for some tests? Btw, slow dispatch perf isn't a priority anymore